src/ruggeduino.cpp

```diff
@@ -47,6 +49,33 @@ void command_mode(int mode) {
   pinMode(pin, mode);
 }
 
+void command_ultrasound() {
+  int pulse = read_pin();
+  int echo = read_pin();
+
+  // config pins to correct modes
+  pinMode(pulse, OUTPUT);
+  pinMode(echo, INPUT);
+
+  // provide pulse to trigger reading
+  digitalWrite(pulse, LOW);
+  delayMicroseconds(2);
+  digitalWrite(pulse, HIGH);
+  delayMicroseconds(5);
+  digitalWrite(pulse, LOW);
+
+  // measure the echo time on the echo pin
+  int duration = pulseIn(echo, HIGH, 60000);
+  Serial.print(microsecondsToMm(duration));
+}
+
+long microsecondsToMm(long microseconds) {
+  // The speed of sound is 340 m/s or 29 microseconds per centimeter.
+  // The ping travels out and back, so to find the distance we need half
+  // 10 x (us / 29 / 2)
+  return (5 * microseconds / 29);
```

Comment:
Thought: Would it be clearer to just combine these lines and avoid the hand-optimising, in hope the compiler will do some of it for us?

```suggestion
  return (10 * microseconds / 29 / 2);
```

Reply:
That's generally bad practice for embedded programming. And when dealing with ints you have to be careful with the order of operations to avoid loss of precision.